src/command_prechecks.ts

```diff
@@ -392,4 +395,103 @@ export default class CommandPrechecks {
 
         return true;
     }
+
+    static answerHiddenPrecheck(precheckArgs: PrecheckArgs): boolean {
+        if (
+            !precheckArgs.session ||
+            precheckArgs.session.isListeningSession() ||
+            (precheckArgs.session as GameSession).gameType !== GameType.HIDDEN
+        ) {
+            return true;
+        }
+
+        if (!precheckArgs.interaction) {
+            if (
+                precheckArgs.parsedMessage!.components.length === 0 ||
+                [AnswerType.TYPING, AnswerType.TYPING_TYPOS].includes(
+                    precheckArgs.parsedMessage!.components[0] as AnswerType
+                )
+            ) {
+                // Allow /answer change to different typing modes during hidden
+                return true;
+            }
+        } else {
+            const { interactionName, interactionOptions } = getInteractionValue(
+                precheckArgs.interaction
+            );
+
+            const action = interactionName as OptionAction;
+            if (action === OptionAction.SET) {
+                if (
+                    [AnswerType.TYPING, AnswerType.TYPING_TYPOS].includes(
+                        interactionOptions["answer"] as AnswerType
+                    )
+                ) {
+                    return true;
+                }
+            }
+        }
+
+        const embedPayload: EmbedPayload = {
+            title: i18n.translate(
+                precheckArgs.messageContext.guildID,
+                "misc.preCheck.title"
+            ),
+            description: i18n.translate(
+                precheckArgs.messageContext.guildID,
+                "misc.preCheck.notHidden"
+            ),
+        };
+
+        sendErrorMessage(
+            precheckArgs.messageContext,
+            embedPayload,
+            precheckArgs.interaction
+        );
+        return false;
+    }
+
+    static timerHiddenPrecheck(precheckArgs: PrecheckArgs): boolean {
+        if (
+            !precheckArgs.session ||
+            precheckArgs.session.isListeningSession() ||
+            (precheckArgs.session as GameSession).gameType !== GameType.HIDDEN
+        ) {
+            return true;
+        }
+
+        if (!precheckArgs.interaction) {
+            if (precheckArgs.parsedMessage!.components.length > 0) {
+                // Allow /timer change but not reset during hidden
```

Comment:
Isn't this the opposite? Also, this comment applies for both interactrion and non interaction branches